pkg/notes/adaptor/repository/prisma.ts

```diff
@@ -0,0 +1,242 @@
+
+  private serialize(note: Note) {
+    const visibility = () => {
+      switch (note.getVisibility()) {
+        case 'PUBLIC':
+          return 0;
+        case 'HOME':
+          return 1;
+        case 'FOLLOWERS':
+          return 2;
+        case 'DIRECT':
+          return 3;
+      }
+    };
+
+    return {
+      id: note.getID() as string,
+      text: note.getContent(),
+      visibility: visibility(),
+      authorId: note.getAuthorID() as string,
+      createdAt: note.getCreatedAt(),
+      deletedAt: Option.isNone(note.getDeletedAt())
+        ? undefined
+        : Option.unwrap(note.getDeletedAt()),
+    };
+  }
+
+  private deserialize(data: DeserializeNoteArgs): Note {
+    const visibility = (): NoteVisibility => {
+      switch (data.visibility) {
+        case 0:
+          return 'PUBLIC';
+        case 1:
+          return 'HOME';
+        case 2:
+          return 'FOLLOWERS';
+        case 3:
+          return 'DIRECT';
+        default:
+          throw new Error('Invalid Visibility');
+      }
+    };
+    return Note.reconstruct({
+      id: data.id as ID<NoteID>,
+      content: data.text,
+      authorID: data.authorId as ID<AccountID>,
+      createdAt: data.createdAt,
+      deletedAt: !data.deletedAt ? Option.none() : Option.some(data.deletedAt),
+      contentsWarningComment: '',
+      originalNoteID: !data.renoteId
+        ? Option.some(data.renoteId as ID<NoteID>)
+        : Option.none(),
+      // ToDo: add SendTo field to schema
+      sendTo: Option.none(),
+      updatedAt: data.updatedAt ? Option.some(data.updatedAt) : Option.none(),
+      visibility: visibility() as NoteVisibility,
+    });
+  }
+
+  async create(note: Note): Promise<Result.Result<Error, void>> {
+    const serialized = this.serialize(note);
+    try {
+      await this.client.note.create({
+        data: serialized,
+      });
+    } catch (e) {
+      return Result.err(e as Error);
+    }
+    return Result.ok(undefined);
+  }
+
+  async deleteByID(id: ID<NoteID>): Promise<Result.Result<Error, void>> {
+    try {
+      // NOTE: logical delete
+      await this.client.note.update({
+        where: {
+          id,
+        },
+        data: {
+          // ToDo: use Note.deletedAt
+          deletedAt: new Date(),
+        },
+      });
+    } catch (e) {
+      return Result.err(e as Error);
+    }
+    return Result.ok(undefined);
+  }
+
+  async findByAuthorID(
+    authorId: ID<AccountID>,
+    limit: number,
+  ): Promise<Option.Option<Note[]>> {
+    try {
+      const res = await this.client.note.findMany({
+        where: {
+          authorId,
+          // NOTE: Exclude from the search those whose deletedAt does not appear undefined.
+          deletedAt: undefined,
+        },
+        take: limit,
+        orderBy: {
+          createdAt: 'desc',
+        },
+      });
+      return Option.some(
+        res.map((v) => this.deserialize(v as DeserializeNoteArgs)),
```

Comment:
```suggestion
        res.map((v) => this.deserialize(v)),
```
これらのキャストは危険なので取り除いても動くようにしたいです. `findByID` も同様です.